### PR TITLE
Auto padding of code blocks

### PR DESCRIPTION
**Pull request by guanglinn**
After using this feature (line numbers) for a long time, I made the following improvements again.
1. Add support for auto left padding of code blocks when enable line numbers;
2. Better margins and padding;
3. Better icon of option (Line numbers);
4. Remove unnecessary extension `flexmark-ext-prism-linenumber`.

### CODE REVIEW

**Comment by guanglinn** [comment]
### Auto left padding of line numbers



### Better margins and padding



### Icon (Line numbers) with more suitable sizes

**Comment by guanglinn** [comment]
Hi, @gsantner . Could I add my name to the list of contributors? 😊

**Comment by guanglinn** [comment]
### The cursor was covered




### Now

**Comment by gsantner** [comment]
How about dark theme? The current color is made to fit both light and dark well enough. Guess the new one does not.

**Comment by guanglinn** [comment]
> How about dark theme? The current color is made to fit both light and dark well enough. Guess the new one does not.


This is in dark theme, is it OK?

File: README.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ Markor is openly developed free software that accepts community contributions.
 Markor supports showing line numbers now. In the top file menu you can find a new option to enable numbers.
 It is supported in editor as well in view mode of documents (in code blocks).
 
-![Line numbers](doc/assets/2023-10-11-line-numbers.webp)
+![Line numbers](doc/assets/2023-10-11-line-numbers.jpg)
```
Comment (gsantner) [inline README.md:55]: Do you mind to convert the new screenshots to webp as well and resize it down to a meaningful size? I usually make it to 600-1000px one dimension.

So far that worked well and saves a lot of unnecessary bytes.

File: app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextConverter.java [hunk 2]
```diff
@@ -160,26 +159,20 @@ public class MarkdownTextConverter extends TextConverterBase {
     @Override
     public String convertMarkup(String markup, Context context, boolean lightMode, boolean lineNum, File file) {
         String converted = "", onLoadJs = "", head = "";
-
         final MutableDataSet options = new MutableDataSet();
 
-        if (lineNum) {
-            // Add code blocks Line numbers extension
-            final ArrayList<Extension> extensions = new ArrayList<>(flexmarkExtensions);
-            extensions.add(LineNumbersExtension.create());
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextConverter.java:169]: So basically this PR removes everything from Java side AST line number parsing and instead line numbers are fully generated at view-time by prism javascript. Right?

File: app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextConverter.java [hunk 3]
```diff
@@ -323,6 +316,11 @@ public String convertMarkup(String markup, Context context, boolean lightMode, b
             }
         }
 
+        if (lineNum) {
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextConverter.java:319]: Can you please rename the bool flag to `enableLineNumbers`?

**Comment by gsantner** [comment]
> This is in the dark theme, is it OK?

To me it's a bit strong contrasted and text bleeding, but somebody else might observe it different.

File: app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextConverter.java (hunk 2 above)
Comment (guanglinn) [reply app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextConverter.java:169]: The work of the extension `flexmark-ext-prism-linenumber` is just to add class `line-numbers` for each code block (`pre`), but I noticed that only add class `line-numbers` to the `body` can also globally enable line numbers, so I removed the extension.

> [Line Numbers ▲ Prism plugins (prismjs.com)](https://prismjs.com/plugins/line-numbers/)

**Comment by guanglinn** [comment]
### Wrap words in code blocks

**Review by gsantner — APPROVED** [review]

**Pull request merged**